### DISCUSSION TITLE
How to count % of used area in Bounding Box?

### DISCUSSION THREAD

**Discussion by dkvadratu** [fabricjs/fabric.js#7028] [Q&A]
Hi.

I need to count % percentage of all objects inside bounding box.  
![fabric-js-elemnt-size-in-bounding-box](https://user-images.githubusercontent.com/7911433/115548231-e6cb6d00-a2af-11eb-9c6b-260798ff9b36.jpg)

With element coordinates it difficult, because object can be rotates. And so on..

Any ideas how to do that? O any person who can help/develop?

I use: WordPress Fancy Product Designer plugin, which uses Fabric.js
I need pricing by used area in Bounding Box.
![price-per-printing-box-usage](https://user-images.githubusercontent.com/7911433/115548296-fea2f100-a2af-11eb-8549-f7bc6202058c.png)

**Comment by dkvadratu** [comment]
I think if I limit that elments can be only in Bounding Box area, then it would be easier to count the size.
Still waiting for opinions.

**Comment by melchiar** [comment]
Measuring the area geometrically is going to be a difficult task, particularly if you're having to account for multiple objects overlapping each other. 

The way I would approach it would be to render an off-screen canvas with a white background, then draw a black rectangle in the size and orientation of each object. Then just do a loop of each pixel in your canvas counting the the ratio of black pixels to white pixels. Not the most elegant solution but I'd guess it would be accurate enough to use for price calculation.

**Comment by timoostrich** [comment]
Some ideas what you can search for:
Rectangle intersection, rotated rectangle intersection, area rotated rectangle intersection
There are some algorithms out there to calculate these kind of things.
